### PR TITLE
feat: add getActivePiecesByCursor for gas-efficient pagination

### LINKED ISSUE #243: `getActivePieces` runs out of gas with high offsets due to O(n) loop from index 0
### Summary

The `getActivePieces` function becomes unusable at higher offsets because the loop always starts from `i = 0`, causing gas costs to scale linearly with offset.

### Reproduction

**Contract:** [0xBADd0B92C1c71d02E7d520f64c0876538fa2557F](https://filecoin.blockscout.com/address/0xBADd0B92C1c71d02E7d520f64c0876538fa2557F?tab=read_write_proxy)

**Dataset:** `setId = 64` (60,000+ pieces on Filecoin mainnet)

| Call | Result |
|------|--------|
| `getActivePieces(64, 10000, 100)` | ✅ Works |
| `getActivePieces(64, 12000, 100)` | ❌ OutOfGas |

### Root Cause
```solidity
for (uint256 i = 0; i < maxPieceId; i++) {  // Always starts from 0
    if (pieceLeafCounts[setId][i] > 0) {
        if (activeCount >= offset && resultIndex < limit) {
            // collect
        }
        activeCount++;
    }
}
```

With offset 12,000, the function performs ~12,000+ cold storage reads just to reach the starting point.

### Suggested Fix

Switch from offset-based to cursor-based pagination using `startPieceId`:
```solidity
function getActivePieces(uint256 setId, uint256 startPieceId, uint256 limit)
```

Callers use the last returned `pieceId` as `startPieceId` for the next page, making pagination O(limit) instead of O(offset).

cc: @rvagg @BigLep @rjan90

### PR DESCRIPTION
**Pull request by Chaitu-Tatipamula**
## Summary
Fixes #243 - Adds `getActivePiecesByCursor` function that uses cursor-based pagination with O(limit) gas complexity, avoiding the gas exhaustion issue with `getActivePieces` at high offsets.

## Changes
- **New function:** `getActivePiecesByCursor(setId, startPieceId, limit)` - starts from `startPieceId` instead of scanning from 0
- **Updated docs:** Added warning to `getActivePieces` about O(offset) gas complexity
- **Tests:** Added 6 new tests covering basic pagination, deleted pieces, edge cases

## Usage
```solidity
// First page
(pieces, ids, hasMore) = getActivePiecesByCursor(setId, 0, 100);

// Next page: use last returned pieceId + 1
(pieces, ids, hasMore) = getActivePiecesByCursor(setId, ids[ids.length-1] + 1, 100);

### CODE REVIEW

File: src/PDPVerifier.sol [hunk 1]
```diff
@@ -391,6 +393,74 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
...
+        if (startPieceId >= maxPieceId) {
+            return (new Cids.Cid[](0), new uint256[](0), false);
+        }
+
+        // Over-allocate arrays to limit size
+        Cids.Cid[] memory tempPieces = new Cids.Cid[](limit);
+        uint256[] memory tempPieceIds = new uint256[](limit);
+        uint256 resultIndex = 0;
+
+        // Start fron startPieceId and collect up to limit
```
Comment (rvagg) [inline src/PDPVerifier.sol]:
```suggestion
        // Start from startPieceId and collect up to limit
```

File: src/PDPVerifier.sol [hunk 2]
```diff
@@ -391,6 +393,74 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
...
+        // Handle case where we found fewer items than limit
+        if (resultIndex == 0) {
+            return (new Cids.Cid[](0), new uint256[](0), false);
+        } else if (resultIndex < limit) {
+            pieces = new Cids.Cid[](resultIndex);
+            pieceIds = new uint256[](resultIndex);
+            for (uint256 i = 0; i < resultIndex; i++) {
+                pieces[i] = tempPieces[i];
+                pieceIds[i] = tempPieceIds[i];
+            }
```
[the remark this answers is not in the document]
Comment (Chaitu-Tatipamula) [reply src/PDPVerifier.sol]: ah alright
